create-calendar-collection.js

```diff
@@ -0,0 +1,280 @@
+        deDuplicatedFileNames.unshift(monthYearPairToPrepend);
+        if (currentMonthToPrepend > 1) {
+            currentMonthToPrepend -= 1;
+        } else {
+            currentMonthToPrepend = 12;
+        }
+        if (currentMonthToPrepend === 12) {
+            currentYearToPrepend -= 1;
+        }
+        --numberOfMonthsToPrepend;
+    }
+    return deDuplicatedFileNames;
+}
+
+/**
+ * Returns the number of days in the month respecting leap years in the case of February where the index is 1.
+ * Note that months are treated as a zero-based index like the JavaScript Date type.
+ * @param {number} year 
+ * @param {number} monthIndex 
+ * @returns {number}
+ */
+function getDaysInMonth(year, monthIndex) {
+    const daysInMonth = [
+        31,
+        year % 4 === 0 ? 29 : 28,
+        31,
+        30,
+        31,
+        30,
+        31,
+        31,
+        30,
+        31,
+        30,
+        31,
+    ];
+    const dayCount = daysInMonth[monthIndex];
+    return dayCount;
+}
+
+/**
+ * Returns the number of days to count back to get from dayName to sunday.
+ * @param {string} dayName Lowercase long name of the weekday.
+ * @returns {number}
+ */
+function getCountOfDaysToSunday(dayName) {
+    const weekdayNames = [
+        'sunday',
+        'monday',
+        'tuesday',
+        'wednesday',
+        'thursday',
+        'friday',
+        'saturday',
+    ];
+    const indexOfDayName = weekdayNames.indexOf(dayName);
+    const dayCount = indexOfDayName > 0 ? -indexOfDayName : 0;
+    return dayCount;
+}
+
+/**
+ * Returns the number of days to count forward to get from dayName to saturday.
+ * @param {string} dayName Lowercase long name of the weekday.
+ * @returns {number}
+ */
+function getCountOfDaysToSaturday(dayName) {
+    const weekdayNames = [
+        'sunday',
+        'monday',
+        'tuesday',
+        'wednesday',
+        'thursday',
+        'friday',
+        'saturday',
+    ];
+    const indexOfDayName = weekdayNames.indexOf(dayName);
+    const dayCount = weekdayNames.length - 1 - indexOfDayName;
+    return dayCount;
+}
+
+/**
+ * Returns a Calendar collection entry file data string that is formatted according to what is expected by the Calendar layout template.
+ * The Calendar collection folder contains a _sample.md file that details the expected Front Matter.
+ * Also the template logic in the calendar layout template may be useful (_layouts/calendar.html).
+ * @param {number[]} yearMonth Array of numbers defining [year, month].
+ * @returns {string}
+ */
+function createCalendarCollectionEntryData(yearMonth) {
+
+    const [year, month] = yearMonth;
+    const monthIndex = month - 1;
+    const previousMonthYear = monthIndex > 0 ? year : year - 1;
+    const previousMonthIndex = monthIndex > 0 ? monthIndex - 1 : 11;
+    const previousMonthEndDay = getDaysInMonth(previousMonthYear, previousMonthIndex);
+    const hoursOffsetBetweenLocalAndUTC = (new Date()).getTimezoneOffset() / 60;
```

Comment:
We should not rely on the local timezone for generating the content to make sure we don't get different content from machines running this script that don't have the same timezone.